src/main/java/sic/service/impl/ProductoServiceImpl.java

```diff
@@ -1130,6 +1130,15 @@ public void getListaDePreciosEnPdf(BusquedaProductoCriteria criteria, long idSuc
             this.getListaDePrecios(productos, FORMATO_PDF), FORMATO_PDF);
   }
 
+  @Override
+  public byte[] getListaDePreciosEnPdf(long[] idProductos) {
+    var productos = new ArrayList<Producto>();
+    for (long idProducto : idProductos) {
+      productos.add(this.getProductoNoEliminadoPorId(idProducto));
```

Comment:
This is very inefficient. The query should be something like SELECT xxxx FROM xxxx WHERE IN .....